File: threads/src/main/java/im/threads/internal/transport/models/Attachment.java
package im.threads.internal.transport.models;

public class Attachment {
    private long id;
    private String result;
    private String name;
    private long size;
    private boolean isSelfie;

    public long getId() {
        return id;
    }

    public String getResult() {
        return result;
    }

    public String getName() {
        return name;
    }

    public long getSize() {
        return size;
    }

    public boolean isSelfie() {
        return isSelfie;
    }
}

File: threads/src/main/java/im/threads/internal/transport/models/Attachment.kt
package im.threads.internal.transport.models

import im.threads.business.models.enums.AttachmentStateEnum
import im.threads.internal.model.ErrorStateEnum

class Attachment(
    val id: Long = 0,
    val result: String? = null,
    val originalUrl: String? = null,
    val name: String? = null,
    val size: Long = 0,
    val type: String? = null,
    var state: AttachmentStateEnum = AttachmentStateEnum.ERROR,
    var errorCode: String? = null,
    val errorMessage: String = ""
) {
    fun getErrorCodeState(): ErrorStateEnum {
        errorCode?.let {
            return ErrorStateEnum.errorStateEnumFromString(it)
        }
        return ErrorStateEnum.ANY
    }
}
